src/snowcli/output/printing.py

```diff
@@ -23,6 +23,75 @@ def default(self, o):
         return super().default(o)
 
 
+class OutputData:
+    def __init__(
+        self,
+        format: Optional[OutputFormat] = None,
+        cursor: Optional[SnowflakeCursor] = None,
+        string_output: Optional[str] = None,
+        data: Optional[List[Dict]] = None,
+    ) -> None:
+        self.output: List[Union[SnowflakeCursor, str, List[Dict]]] = []
+        self.counter = 0
+
+        if format:
+            self.format = format
+        else:
+            self.format = _get_format_type()
+
+        if cursor:
+            self.add_cursor(cursor)
+
+        if string_output:
+            self.add_string_output(string_output)
+
+        if data:
+            self.add_data(data)
+
+    def add_cursor(self, cursor: SnowflakeCursor):
+        self.output.append(cursor)
```

Comment:
Should we do type validation in every method? We don't want users to add cursors via `add_data`, right?

Reply:
That was not necessary, merged it into one method

Reply:
But this is stil case right? What will happen if someone passes something that we do not accept?

Reply:
You mean to check if data is one of type [str, List[Data], SnowflakeCursor] with isinstance?